source/common/json/config_schemas.cc

```diff
@@ -448,6 +448,7 @@ const std::string Json::Schema::ROUTE_ENTRY_CONFIGURATION_SCHEMA(R"EOF(
       "prefix" : {"type" : "string"},
       "path" : {"type" : "string"},
       "cluster" : {"type" : "string"},
+      "cluster_header" : {"type" : "string"},
       "weighted_clusters": {"$ref" : "#/definitions/weighted_clusters"},
       "host_redirect" : {"type" : "string"},
```

Comment:
I prefer we keep the code checks in place. They are easy to follow. If we were to move this to schema check, it would be hard to read and a lot harder to validate a correct schema without adding more tests.